core/region.go

```diff
@@ -206,31 +206,30 @@ func (r *region) determineInstanceTypeInformation(cfg *Config) {
 		price.spot = make(spotPriceMap)
 		price.ebsSurcharge = it.Pricing[r.name].EBSSurcharge
 
-		// if at this point the instance price is still zero, then that
-		// particular instance type doesn't even exist in the current
-		// region, so we don't even need to create an empty spot pricing
-		// data structure for it
-		if price.onDemand > 0 {
-			// for each instance type populate the HW spec information
-			info = instanceTypeInformation{
-				instanceType:        it.InstanceType,
-				vCPU:                it.VCPU,
-				memory:              it.Memory,
-				GPU:                 it.GPU,
-				pricing:             price,
-				virtualizationTypes: it.LinuxVirtualizationTypes,
-				hasEBSOptimization:  it.EBSOptimized,
-			}
+		if price.onDemand == 0 {
```

Comment:
I'm aware it's unlikely to happen, but maybe switching to `<= 0` would be safer as even if the value is negative the code would warn us.

Also according to the previous method's behaviour, shouldn't this be followed by an `else` afterwards? Or a `return` on the `if` itself if no else is desired?

Reply:
This should never be negative, the check is for the zero value in case we're missing the data.

I'll revisit the if logic on Friday.

Reply:
I agree with @xlr-8's question and suggestion.